GG/GG/Font.h

```diff
@@ -233,6 +241,40 @@ class GG_API Font
         mutable X cached_width;
     };
 
+    /** \brief Used to encapsulate a text and tokens together to be rendered
+        using GG::Font. */
+    class GG_API TextElements
+    {
+    public:
+        TextElements(const Font& font);
+        ~TextElements();
+
+        /** Return the constructed text.*/
+        const std::string& Text();
+        /** Return the constructed TextElements.*/
+        const std::vector<boost::shared_ptr<TextElement> >& Elements();
```

Comment:
As far a I know the vector of TextElement is always flat.  However, when it is doing pre-rendering it uses a stack to maintain hierarchical information of which tags are active. 

For your example the vector of TextElement, listed from top to bottom, with one TextElement per line, would be:
- TEXT text = "The"
- WHITESPACE text = " " 
- OPEN_TAG text = "<i>", tag_name = "i"
- TEXT text = "quick" 
- WHITESPACE text = " " 
- OPEN_TAG text = "<rgba 102 51 0 255>" tag_name = "rgba",  four params = 102,51,0,255
- TEXT text = brown
- CLOSE_TAG text = </rgba>, tag_name = rgba
- CLOSE_TAG, text = </i> tag_name = i
- TEXT text = "fox" 
- WHITESPACE text = " " 

etc.

I've changed the poorly chosen name of `TextElements` to `TextAndElementsAssembler`, which clarifies that it constructs a matched pair of Text and a vector of TextElements.